libsplinter/src/admin/client/event/mod.rs

```diff
@@ -93,6 +94,15 @@ pub trait AdminServiceEventClient {
     ///
     /// This should block until an event is available.
     fn next_event(&self) -> Result<AdminServiceEvent, NextEventError>;
+
+    /// Wait for the given event up until a timeout
+    ///
+    /// This should block until an event is available.
+    fn wait_for(
+        &self,
+        event_type: &EventType,
```

Comment:
This argument needs to contain more detail then just the type.  More likely then not, the test will be waiting on a specific circuit change.

Perhaps an enum like
```rust
enum EventQuery {
    CircuitReady(String), // or a struct variant with circuit id
    CircuitRejected(String),
    CircuitDisbanded(String),
    // ... etc
    // The voting ones would be similar but include the PublicKey
}
```

Reply:
That's a good point.

I created an `EventQuery` enum that's consumed by `wait_for` and uses the filters you list above. To cover all possible uses cases, `wait_for` calls a new function `wait_for_filter` that will accept any arbitrary closure to act as an event filter.

Under the hood, the `EventQuery` enum just transforms itself into an event filter closure.